src/BasicInputs/BasicNumberInput.js

```diff
@@ -51,4 +63,36 @@ BasicNumberInput.propTypes = {
   labelProps: PropTypes.object
 };
 
+function NumberFormatCustom(props) {
+  const { inputRef, onChange, max, min, ...other } = props;
+
+  const withValueLimit = (inputObj) => {
+    const { value } = inputObj;
+    if (min <= value && value <= max) return inputObj;
+  };
+
+  return (
+    <NumberFormat
+      {...other}
+      getInputRef={inputRef}
+      isAllowed={withValueLimit}
+      onValueChange={(values) => {
+        onChange({
+          target: {
+            name: props.name,
+            value: values.value,
+          },
+        });
+      }}
+      isNumericString
+    />
+  );
+}
+
+NumberFormatCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+  name: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired
+};
+
```

Comment:
In order to keep one file per component, can we move this part to a separate file? (in a "src/BasicInputs/components" folder)